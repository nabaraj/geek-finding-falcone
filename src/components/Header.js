import React, { Component } from "react";
import { connect } from "react-redux";
import { resetAppFn } from "./../actions/getDataAction";
import { bindActionCreators } from "redux";

import Notification from "./Notification";
class Header extends Component {
  render() {
    return (
      <header className="App-header shadow-sm p-3 mb-5 bg-white">
        <div className="container">
          <ul className="nav justify-content-end headerNav">
            <li>
              <span
                onClick={e => this.props.resetAppFn(true)}
                className="text-success resetLink"
              >
                Reset
              </span>
            </li>
            <li>
              <a href="https://www.geektrust.in/" className="text-success">
                Geek Trust Home
              </a>
            </li>
          </ul>
        </div>
        {this.props.notification && (
          <Notification notification={this.props.notification} />
        )}
      </header>
    );
  }
}
function mapStateToProps(state) {
  return {
    notification: state.appsData.notificationObject
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      resetAppFn: resetAppFn
    },
    dispatch
  );
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
