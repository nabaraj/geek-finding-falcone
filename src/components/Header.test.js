import React from "react";
import configureStore from "redux-mock-store";
import { mount } from "enzyme";

import Header from "./Header";

const middlewares = [];
const mockStore = configureStore(middlewares);
let initialData = {};
initialData["appsData"] = {};

const dummyProps = {
  notification: "error",
  resetAppFn: "fn()"
};

const store = mockStore(initialData);
describe("Test AppContainer", function() {
  const wrapper = mount(<Header store={store} {...dummyProps} />);
  it("match snapeshots", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("check header element", () => {
    expect(wrapper.find(".App-header").length).toBe(1);
  });
});
