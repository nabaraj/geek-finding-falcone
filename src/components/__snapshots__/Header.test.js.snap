// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppContainer match snapeshots 1`] = `ReactWrapper {}`;
